modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaProductDTOConverterReferenceCheck.java

```diff
@@ -0,0 +1,92 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.source.formatter.check;
+
+import com.liferay.petra.string.StringBundler;
+import com.liferay.petra.string.StringPool;
+import com.liferay.source.formatter.parser.JavaClass;
+import com.liferay.source.formatter.parser.JavaClassParser;
+import com.liferay.source.formatter.parser.JavaTerm;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Kyle Miho
+ */
+public class UpgradeJavaProductDTOConverterReferenceCheck
+	extends BaseUpgradeCheck {
+
+	@Override
+	protected String format(
+			String fileName, String absolutePath, String content)
+		throws Exception {
+
+		if (!fileName.endsWith(".java")) {
+			return content;
+		}
+
+		JavaClass javaClass = JavaClassParser.parseJavaClass(fileName, content);
+
+		for (JavaTerm childJavaTerm : javaClass.getChildJavaTerms()) {
+			if (childJavaTerm.hasAnnotation("Reference") &&
+				childJavaTerm.getContent(
+				).contains(
```

Comment:
I think we should avoid chaining in this case.

Reply:
Like this? 
`
String content = childJavaTerm.getContent();

if (childJavaTerm.hasAnnotation("Reference") &&
	content.contains("ProductDTOConverter")) {
`

Or is there a better way because I thought we werent supposed to declare a variable for a single use immediately after?

Reply:
From my experience, "avoid chaining" overrides "avoid single-use variable".

Reply:
@drewbrokke fyi I resent with avoid chaining